### ISSUE TITLE
A lot of unexpected restarts

### ISSUE THREAD

**Issue by rflemming** [vitotai/BrewPiLess#55]
I just finished building an ispindel so I was testing out integration with brewpiless (using SoftAP mode) and I've been seeing a lot of random restarts. I'm assuming they are caused by the watchdog timer kicking in for some reason or another.  I don't believe they are related to ispindel specifically since I'm pretty sure they happened in the past, just maybe not with the same frequency.  You can see all the resume events on the chart:

![bpl](https://user-images.githubusercontent.com/7843512/34596950-997604fa-f198-11e7-88f3-b057c8770a52.png)

Is there anything I can do to help debug this further?  I tried keeping a window open with the log from testcmd.htm but I haven't been able to capture anything that appears useful.  I'm happy to help I just don't know what sort of info would be beneficial.

**Comment by vitotai** [comment]
I don't know either.
The only way I would suggest is to check your power supply and ESP8266.

A friend on HBT had the same issue, and he ended up replacing his NodeMcu board to get good result.

It's funny that I was thinking about exception handling of ESP8266. Unfortunately, the source code is not accessible, and we have nothing we can do.

**Comment by quimnut** [comment]
esp8266's have been said to use up to 850mA of 3.3v which is also about what lm1117's are rated to, you can try moving the relay power off the 3.3v pin if you do that.

**Comment by quimnut** [comment]
On the topic of restarts, I upgraded to 2.4.2 and connected an iSpindel yesterday.

![ispindelrestarts](https://user-images.githubusercontent.com/2128947/34920856-66afdb6a-f9c5-11e7-8c46-7cf9301083a8.JPG)

**Comment by rflemming** [comment]
Yup, pretty much what I was seeing.  Did you by chance update to 2.4.2 via the file upload or did you do it over USB?  I did originally did it via the OTA update, but yesterday I decided to pull it out, compile my own 2.4.2 and reflash it via USB.  So far I don't see any resume events in the UI, but I'm also not willing to say they aren't happening anymore yet either.

The problem I'm seeing now (though evidently it's been happening for almost 3 weeks) is that updates aren't showing up at Ubidots.  Nothing changed on my side, so I'm not sure what's going on and I haven't spent a lot of time trying to debug it yet.

**Comment by quimnut** [comment]
Yes I did, I just put a 100uF cap on the 3v3 pin to see if it was noise/power but it's no different. It's taking 3 attempts to just get the web interface to load.

Was my first go at using the ispindel yesterday but I noticed if I didn't reset then go in via softap and save the settings it wouldn't start sending data. If I just turned it on after saving things previously it wouldn't do anything obvious, With the softap/save method i see it renew it's dhcp lease every upload interval.

**Comment by quimnut** [comment]
usb flashing has fixed the stability for me too. Now to figure out the rest :)

**Comment by rflemming** [comment]
Interesting that flashing via USB seems to have helped you as well.  I'm not seeing the resets anymore, but the web UI does still seem to hang every now and then.  As for my issue with Ubidots, that's working again as well.  I suspect disabling and re-enabling may have done the trick.

**Comment by BB-grooves** [comment]
Are you using a OLED display?

with latest firmware there is not a lot of heapspace - when it gets low you will experience no responses on the webserver and restarts as the watchdog reboots the ESP8266.

For OLED add -DOLEDDISPLAY_REDUCE_MEMORY=true to your build

heap space is low even without the 8k / 16k for buffering the OLED. So try reducing the various buffer sizes from 2048 & 1024 to 512 - however this might cause other issues for large configurations/profiles. I'm currently trying to find optimal values.

**Comment by rflemming** [comment]
Nope, LCD for me.

**Comment by vitotai** [comment]
FYI, you can check the heap status by this URL, like
http://brewpiless.local/fs

I would suggest to get another D1 mini or NodeMcu. They are cheap and people have bad copies from time to time. (Luckily, I don't.)

**Comment by roguenorman** [comment]
my setup was working for a week and then all of a sudden it started to restart every 1min 30 seconds-ish. I am using a D1 mini and also tried a spare that I have but I was getting the same results. I was on version 3.4. I checked the heap size and it is around 16000. I ended up disabling the LCD, buttons and the buzzer but I'm not sure if that would lower heap size. I used 3.5 and finally got it to stay on for 10 minutes so my cooling cycle would kick in. is 16000 heap a lot? I had it set to fridge const and it seemed stable but when i set it to beer const that is when the issues started. Should that matter? or is it coincidence. Is it even the heap making it reset?? I would not even care that much about restarts but it makes the timer start again so the minCoolIdleTime gets reset everytime so i pretty much never get a cooling cycle

**Comment by quimnut** [comment]
I would try formatting spiffs if you have not already.

**Comment by roguenorman** [comment]
Yeah actually I did do that just before I got it working again. I lost track of what changes I was making after a while but that could definitely be the thing that fixed it.
